components/brave_wallet/browser/nft_metadata_fetcher.cc

```diff
@@ -225,16 +226,17 @@ void NftMetadataFetcher::FetchMetadata(
 
 void NftMetadataFetcher::OnSanitizeTokenMetadata(
     GetTokenMetadataIntermediateCallback callback,
-    data_decoder::JsonSanitizer::Result result) {
+    api_request_helper::ValueOrError result) {
   if (!result.has_value()) {
     VLOG(1) << "Data URI JSON validation error:" << result.error();
     std::move(callback).Run(
         "", static_cast<int>(mojom::JsonRpcError::kParsingError),
         l10n_util::GetStringUTF8(IDS_WALLET_PARSING_ERROR));
     return;
   }
-
-  std::move(callback).Run(*result, 0, "");  // 0 is kSuccess
+  std::string json;
+  base::JSONWriter::Write(std::move(result).value(), &json);
```

Comment:
This doesn't add extra overhead because `JsonSanitizer::Sanitize` did the same thing internally:
https://source.chromium.org/chromium/chromium/src/+/main:services/data_decoder/public/cpp/json_sanitizer_non_android.cc;l=22

Better to refactor the code to us `base::Value` directly (instead converting it back to string).

Reply:
Needs a todo like below to refactor this 
`// TODO(apaymyshev): parse metadata in wallet's backend`

Reply:
done